collectors/python.d.plugin/anomalies/README.md

```diff
@@ -0,0 +1,202 @@
+<!--
+---
+title: "anomalies"
+custom_edit_url: https://github.com/netdata/netdata/edit/master/collectors/python.d.plugin/anomalies/README.md
+---
+-->
+
+# Anomalies - ML driven anomaly detection for your Netdata!
+
+This collector uses the Python [PyOD](https://pyod.readthedocs.io/en/latest/index.html) library to perform unsupervised [anomaly detection](https://en.wikipedia.org/wiki/Anomaly_detection) on your Netdata charts and/or dimensions.
+
+Instead of this collector just _collecting_ data, it also does some computation on the data it collects to return an anomaly probability and anomaly flag for each chart or custom model you define. This computation consists of a **train** function that runs every `train_n_secs` to train the ML models to learn what 'normal' typically looks like on your node. At each iteration there is also a **predict** function that uses the latest trained models and most recent metrics to produce an anomaly probability and anomaly flag for each chart or custom model you define. 
+
+## Charts
+
+Two charts are produced:
+
+- **Anomaly Probability** (`anomalies.probability`): This chart shows the probability that the latest observed data is anomalous based on the trained model for that chart (using the [`predict_proba()`](https://pyod.readthedocs.io/en/latest/api_cc.html#pyod.models.base.BaseDetector.predict_proba) method of the trained PyOD model).
+- **Anomaly** (`anomalies.anomaly`): This chart shows `1` or `0` predictions of if the latest observed data is considered anomalous or not based on the trained model (using the [`predict()`](https://pyod.readthedocs.io/en/latest/api_cc.html#pyod.models.base.BaseDetector.predict) method of the trained PyOD model).
+
+Below is an example of the charts produced by this collector and how they might look when things are 'normal' on the node. The anomaly probabilities tend to bounce randomly around a typically low probability range, one or two might randomly jump or drift outside of this range every now and then and show up as anomalies on the anomaly chart. 
+
+![alt text](https://github.com/andrewm4894/random/blob/master/images/netdata/netdata-anomalies-collector-normal.jpg)
```

Comment:
Can you add some alt text to each of the images? It's good for SEO and accessibility purposes.

Also, I'd recommend you upload the files to GitHub's static CDN (attach them to comment and copy the URLs), so that we know they stay available and aren't directly connected to your GitHub account/repo.

Reply:
Will do - ill attach them in here and use that link

![netdata-alarms-collector](https://user-images.githubusercontent.com/2178292/100630484-8482c780-3322-11eb-920b-3e0018f4056e.jpg)
![netdata-anomalies-collector-abnormal](https://user-images.githubusercontent.com/2178292/100630489-851b5e00-3322-11eb-87e5-509ca63e7afd.jpg)
![netdata-anomalies-collector-normal](https://user-images.githubusercontent.com/2178292/100630492-85b3f480-3322-11eb-842a-cd3e5a5917ec.jpg)
![netdata-anomalies-collector-normal-again](https://user-images.githubusercontent.com/2178292/100630494-85b3f480-3322-11eb-94cc-9f94ced3acb1.jpg)
![netdata-anomalies-online-collector-abnormal](https://user-images.githubusercontent.com/2178292/100630497-864c8b00-3322-11eb-98c2-54a5f4d98f59.jpg)
![netdata-anomalies-online-collector-normal](https://user-images.githubusercontent.com/2178292/100630499-864c8b00-3322-11eb-9bb4-4b41c1832850.jpg)
![netdata-anomalies-online-collector-normal-again](https://user-images.githubusercontent.com/2178292/100630502-86e52180-3322-11eb-9742-de75ac8eedcd.jpg)
![netdata-zscores-collector-abnormal](https://user-images.githubusercontent.com/2178292/100630503-86e52180-3322-11eb-874c-a2f3841d4256.jpg)
![netdata-zscores-collector-normal](https://user-images.githubusercontent.com/2178292/100630504-877db800-3322-11eb-9898-cba0a3cad564.jpg)
![netdata-zscores-collector-normal-again](https://user-images.githubusercontent.com/2178292/100630506-88164e80-3322-11eb-8f81-aad63aaa0119.jpg)